elastic/kibana
# [Infra UI] Fix broken Hosts View UI in serverless

fixes https://github.com/elastic/kibana/issues/167230

## Summary

This PR fixes the sticky search bar and the position of the popover on the hosts view page  in serverless mode

**Serverless**

https://github.com/elastic/kibana/assets/2767137/636af516-7f86-4bb2-b24a-c679d508ebfd

<img width="1686" alt="image" src="https://github.com/elastic/kibana/assets/2767137/de5a78a3-a7f6-4861-a8fc-304f5dee40d9">

**Stateful**

https://github.com/elastic/kibana/assets/2767137/385378b2-fbae-43ea-93c2-84ad6d7e6afc

<img width="1714" alt="image" src="https://github.com/elastic/kibana/assets/2767137/62190221-8a69-4a73-8a7f-e0d7f088d8d2">


For the sticky header, the `useKibanaHeader` hooks was changed to account only for the sticky action menu height, that's present in serverless. We can rely on the css var `--euiFixedHeadersOffset` for the main offset between main header and page content.

### How to test
- Start a local es instance: `yarn es serverless --kill --clean --license trial --ssl`
- Enable `infra` in the `serverless.oblt.dev.yml` file:
  - `xpack.infra.enabled: true`
- Start a local kibana instance: `yarn serverless-oblt --ssl` and see if the side nav contains the Infrastructure item
- Navigate to `https://0.0.0.0:5601/ftw/app/hosts` and check the table popovers and sticky search bar.
